### DISCUSSION TITLE
mkdocs-material/blog/2023/09/22/using-git-sparse-checkout-for-faster-documentation-builds/

### DISCUSSION THREAD

**Discussion by giscus** [squidfunk/mkdocs-material#6066] [_]
# mkdocs-material/blog/2023/09/22/using-git-sparse-checkout-for-faster-documentation-builds/

Write your documentation in Markdown and create a professional static site in minutes – searchable, customizable, in 60+ languages, for all devices

https://squidfunk.github.io/mkdocs-material/blog/2023/09/22/using-git-sparse-checkout-for-faster-documentation-builds/

**Comment by Andre601** [comment]
I assume I could use this in other CIs too such as Woodpecker-CI?

The reason I ask is, that Woodpecker-CI is by default doing a shallow fetch of data, so I do a `git fetch --unshallow` here, which also includes unnecessary data such as git tags, so I wonder if the `sparse-checkout` would be an alternative to utilize here, or if I would need to use a different solution with `git fetch`.

Would be cool if other CI/CD examples could be added in the future (If doable of course).

**Reply by squidfunk** [reply]
Yes, it will always work if you use `git`, although I have no idea how to configure it for the provider you mentioned. For GitHub Actions, it's just a simple config setting, but you can certainly just use the `git` CLI to achieve the same.
